src/main/java/uk/gov/hmcts/reform/professionalapi/resolver/HeaderVersionArgumentResolver.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.professionalapi.resolver;
+
+import javax.servlet.http.HttpServletRequest;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.MethodParameter;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import uk.gov.hmcts.reform.auth.checker.spring.serviceanduser.ServiceAndUserDetails;
+
+
+@Slf4j
+public class HeaderVersionArgumentResolver implements HandlerMethodArgumentResolver {
+
+    @Override
+    public boolean supportsParameter(MethodParameter methodParameter) {
+
+        return methodParameter.getParameterAnnotation(OrgId.class) != null;
+    }
+
+    @Override
+    public Object resolveArgument(
+            MethodParameter methodParameter,
+            ModelAndViewContainer modelAndViewContainer,
+            NativeWebRequest nativeWebRequest,
+            WebDataBinderFactory webDataBinderFactory) throws Exception {
+
+        HttpServletRequest request
+                = (HttpServletRequest) nativeWebRequest.getNativeRequest();
+        String userId = null;

```

Comment:
Don't need to assign userId to null. Just use String userId;

Reply:
if null not assigned then it will give compilation error or i need to write two return statements.
To avoid that i have assigned null in the initialization

Reply:
You could refactor the code to a private method that checks if serviceAndUserDetails are present and returns null if they are not and just return the result. This would be more readable in my opinion.